src/accsr/remote_storage.py

```diff
@@ -460,34 +476,62 @@ def push(
 
         Note that ``path`` may not be absolute if ``local_path_prefix`` is specified.
 
-        Remote objects will not be overwritten if their MD5sum matches the local file.
-
         :param path: Path to the local object (file or directory) to be uploaded, may be absolute or relative
         :param local_path_prefix: Prefix to be concatenated with ``path``
-        :param overwrite_existing: If a remote object already exists, overwrite it?
+        :param force: If False push will raise an error if an already existing remote file deviates from the local
+        in it's md5sum. If True these files are overwritten.
         :param path_regex: If not None only files with paths matching the regex will be pushed
+        :param dryrun: If True simulates the pull operation and the returns local the objects that would have been pulled.
         :return: A list of :class:`Object` instances for all remote objects that were created or matched existing files
         """
         local_path = self._get_push_local_path(path, local_path_prefix)
-        if os.path.isfile(local_path):
+        summary = self.collect_push_summary(path, local_path_prefix, path_regex)
 
-            if path_regex is not None and not path_regex.match(path):
-                log.warning(
-                    f"{path} does not match regular expression '{path_regex}'. Nothing is pushed."
-                )
-                return []
-
-            return [self.push_file(path, local_path_prefix, overwrite_existing)]
+        new_files = summary["new_files"]
+        existing_eq_md5 = summary["existing_eq_md5"]
+        existing_neq_md5 = summary["existing_neq_d5"]
+        existing = existing_neq_md5 + existing_eq_md5
 
-        elif os.path.isdir(local_path):
-            return self.push_directory(
-                path, local_path_prefix, overwrite_existing, path_regex=path_regex
+        if not force and len(existing_neq_md5) > 0:
+            log.error(
+                "Found existing files that would have been overwritten."
+                "Set force=True to allow accsr to overwrite the files"
             )
-        else:
-            raise FileNotFoundError(
-                f"Local path {local_path} does not refer to a file or directory"
+            raise FileExistsError(
+                f"Found {len(existing_neq_md5)} already existing file(s)."
             )
 
+        upload_list = new_files + existing_neq_md5
+        upload_size = sum([obj.size for obj in upload_list])
+
+        if dryrun:
```

Comment:
dryrun hadling happens after check for force now, so an error can be raised even if dryrun is true. I wouldn't expect that. 

Proposal: if dryrun=True and a pull/push would fail b/c of force=False, a message is logged on warning saying what would go wrong. The file collisions can then be read off the summary. This would then actually the suggested way to find such collisions

Reply:
This is connected with the return type of the functions. Maybe we should discuss tomorrow.

Reply:
Let's discuss tomorrow. Although to me it seems like merely moving this block of code higher up and adding a log, but I guess I am missing something